### PR TITLE
Fix homepage URL rendering in client list

### LINKED ISSUE #3057: Homepage URL is wrong in Client list in ClientSection
### Describe the bug

If a client configures a `rootUrl` and relative `baseUrl`, then the homepage URL in the client list is wrong.

With Keycloak running on `http://localhost:8080` and the following client configuration:
```
clientId: test-client
rootUrl: http://localhost:4200
baseUrl: /home
```

image

Yields the following url:

image

-> `/home` resolves to `http://localhost:8080/home` instead of `http://localhost:4200/home`.

### Version

19.0.1

### Expected behavior

Client list should render the correct URL

### Actual behavior

The homepage link should consider the rootUrl and baseUrl correctly.

### How to Reproduce?

1) Create a client with a rootUrl and relative baseUrl.
2) Check the homepage link on the client list.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by thomasdarimont**
Closes #3057

### CODE REVIEW

**Comment by jonkoops** [comment]
@thomasdarimont could you point towards the issue this PR resolves or add a description to the PR?

File: src/clients/ClientsSection.tsx [hunk 1]
```diff
@@ -103,6 +103,40 @@ export default function ClientsSection() {
...
+      return baseUrl;
+    }
+
+    if (rootUrl.startsWith("http")) {
+      let targetUrl = rootUrl;
+      if (baseUrl) {
+        targetUrl = addTrailingSlash(targetUrl) + baseUrl.substring(1);
+      }
+      return targetUrl;
+    }
```
Comment (edewit) [inline src/clients/ClientsSection.tsx:135]:
```suggestion
    if (rootUrl.startsWith("http") && baseUrl) {
        return addTrailingSlash(rootUrl) + baseUrl.substring(1);
   }
   
   return rootUrl;
```

Comment (edewit) [reply src/clients/ClientsSection.tsx:135]: but shouldn't we test that baseUrl starts with a slash instead of always doing a `baseUrl.substring(1)`?

Comment (jonkoops) [reply src/clients/ClientsSection.tsx:135]: That's why I created the new [`joinPath`](https://github.com/keycloak/keycloak-admin-ui/blob/main/src/utils/joinPath.ts) utility :wink:

File: src/clients/ClientsSection.tsx [hunk 2]
```diff
@@ -103,6 +103,40 @@ export default function ClientsSection() {
     </TableText>
   );
 
+  const ClientHomePage = (client: ClientRepresentation) => {
```
Comment (jonkoops) [inline src/clients/ClientsSection.tsx:106]: Can we extract this logic somewhere and add some unit tests? With the amount of branches in this function it feels hard to refactor without accidentally making breaking changes.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/clients/ClientsSection.tsx (hunk 2 above)
Comment (edewit) [reply src/clients/ClientsSection.tsx:106]: :+1: seems that @thomasdarimont is already unsure if all cases are met...

**Review by jonkoops — APPROVED** [review]
Let's merge this now as it fixes an actual bug and create a task to add some tests for this later.

File: src/clients/ClientsSection.tsx [hunk 3]
```diff
@@ -103,6 +103,40 @@ export default function ClientsSection() {
...
+
+    if (rootUrl === "${authBaseUrl}" && baseUrl) {
+      // eslint-disable-next-line prettier/prettier
+      return rootUrl.replace("${authBaseUrl}", addTrailingSlash(adminClient.baseUrl)) + baseUrl.substring(1);
+    }
+
+    if (rootUrl === "${authAdminUrl}" && baseUrl) {
+      // eslint-disable-next-line prettier/prettier
+      return rootUrl.replace("${authAdminUrl}", addTrailingSlash(adminClient.baseUrl)) + baseUrl.substring(1);
+    }
```
Comment (edewit) [inline src/clients/ClientsSection.tsx:123]: If I read this correctly it means when `baseUrl` is not set and rootUrl contains either `"${authBaseUrl}"` or `"${authAdminUrl}"` then we return `undefined`? @thomasdarimont

**Pull request merged**